### ISSUE TITLE
create a zoom_to_aoi btn on the map 

### ISSUE THREAD

**Issue by 12rambau** [12rambau/sepal_ui#389]
It could be a simple button that once set zoom to specific coordinates. (I have a terrible mouse and I get lost quite often)

From the dev point of view it's simply a value that need to be updated in the map when a layer is added.

I don't know if it could b useful from sepal-ui or if it's just something I should implement where I need it.

**Comment by dfguerrerom** [comment]
I think that it could be useful, especially if the given aoi is small, changes to get lost... however, what if you have an AOI with several far away features/geometries? the zoom will point to nowhere and perhaps can cause confusion, and if there are more than one layer? the button will allow to select the 'aoi' layer?

**Comment by 12rambau** [comment]
I was thinking about something like this: 

![aoi_selector](https://user-images.githubusercontent.com/12596392/152364527-0561841b-6f31-4f71-a710-9f3b9ea86929.png)

in the python code that would look like the following :

```python
m = sepal_ui.mapping.SepalMap()

m.add_layer(layer) # add the layer to the map 
m.zoom_bounds(layer.bounds) # zoom on the appropriate geometry (e.g. the AOI)
m.add_aoi(layer.geometry()) # the geometry for the zoom button 
```

## Use cases

### the geometry is unique 

Then the button has no hover panel, simply click on it and we apply zoom_bounds on the referenced geometry

### the geometry has multiple feature 

Then the button has 2 behaviours: 

1. click on the btn to applys zoom_bounds on the `total_bounds`member (assuming it's a geopandas dataset
2. hover the btn to see the lis of the sub_aoi and click on the appropiate one

## What we can't do

digest a list of geometry, the developer need to choose what an aoi to zoom to. the advantage being that the AOI is not linked to the layer but an outside geometry (that can be completely unrelated). it can come very handy for points viz for example.

**Comment by dfguerrerom** [comment]
Ok, I like the proposed UI implementation, it seems to be clear for the end-user. 

So at the end, every layer will have its own custom developer-provided bounds?.

**Comment by 12rambau** [comment]
I would rephrase it to: "every map could have its own developer provided bounds". 

Meaning that if nothing is set, it's simply not displayed and if multiple layers are on the map only the developer-provided bounds will be taken into account.
